awscli/customizations/s3/s3handler.py

```diff
@@ -523,6 +525,118 @@ def _process_transfer(self, future):
             return CommandResult(1, 0)
 
 
+class S3TransferHandlerFactory(object):
+    def __init__(self, cli_params, runtime_config):
+        """Factory for S3TransferHandlers
+
+        :type cli_params: dict
+        :param cli_params: The parameters provide to the CLI command
+
+        :type runtime_config: RuntimeConfig
+        :param runtime_config: The runtime config for the CLI command
+            being run
+        """
+        self._cli_params = cli_params
+        self._runtime_config = runtime_config
+
+    def __call__(self, client, result_queue):
+        """Creates a S3TransferHandler instance
+
+        :type client: botocore.client.Client
+        :param client: The client to power the S3TransferHandler
+
+        :type result_queue: queue.Queue
+        :param result_queue: The result queue to be used to process results
+            for the S3TransferHandler
+
+        :returns: A S3TransferHandler instance
+        """
+        transfer_config = create_transfer_config_from_runtime_config(
+            self._runtime_config)
+        transfer_manager = TransferManager(client, transfer_config)
+
+        LOGGER.debug(
+            "Using a multipart threshold of %s and a part size of %s",
+            transfer_config.multipart_threshold,
+            transfer_config.multipart_chunksize
+        )
+        result_recorder = ResultRecorder()
+        result_processor_handlers = [result_recorder]
+        self._add_result_printer(result_recorder, result_processor_handlers)
+        result_processor = ResultProcessor(
+            result_queue, result_processor_handlers)
+        command_result_recorder = CommandResultRecorder(
+            result_queue, result_recorder, result_processor)
+
+        return S3TransferHandler(
+            transfer_manager, self._cli_params, command_result_recorder)
+
+    def _add_result_printer(self, result_recorder, result_processor_handlers):
+        result_printer = None
+        if self._cli_params.get('quiet'):
```

Comment:
I wonder if this warrants having a ``ResultPrinterFactory``? It is near impossible to test this without patching or running somewhat of a functional test by seeing what gets printed out when the s3 transfer handler is called. We do have functional tests for this so I do not feel as bad for not having a unit test for this specific logic, but would be nice to unit test it.